### PR TITLE
Feature/filter get purchase requests

### PR DESCRIPTION
**Pull request by Gonzanic97**

### CODE REVIEW

File: src/api/handlers.clj [hunk 1]
```diff
@@ -1,30 +1,80 @@
...
+  (if (>= (persistence/get-remaining-budget (:user-email params) (t/year (t/today))) (:price params))
+    (do (persistence/create-purchase-request (assoc params :timestamp (t/today)))
         (response/ok))
 
     (response/bad-request {:error "Remaining budget is not sufficient to cover this purchase"})))
 
+(defn filter-purchase-requests [purchase-requests query-params]
+  (let [state (:state query-params)
+        user-name (:user-name query-params)]
+    (println query-params)
```
Comment (DTissot) [inline src/api/handlers.clj:26]: We should probably get rid of these println calls

File: src/api/handlers.clj [hunk 2]
```diff
@@ -1,30 +1,80 @@
...
+        purchase-requests)
+      )))
 
-(defn get-purchase-requests []
-  (let [purchase-requests (db/get-purchase-requests)]
-    (if (= 0 purchase-requests) (response/no-content) (response/ok {:purchase-request purchase-requests}))))
+(defn get-purchase-requests [{:keys [query-params-map]}]
+  (let [purchase-requests (persistence/get-purchase-requests)
+        filtered-requests (filter-purchase-requests purchase-requests query-params-map)]
+    (println query-params-map)
```
Comment (DTissot) [inline src/api/handlers.clj:42]: Another println

File: src/api/handlers.clj (hunk 1 above)
Comment (Gonzanic97) [reply src/api/handlers.clj]: just removed

File: src/api/handlers.clj (hunk 2 above)
Comment (Gonzanic97) [reply src/api/handlers.clj]: just removed

**Pull request merged**